app/models/vm_or_template.rb

```diff
@@ -1431,10 +1431,8 @@ def v_is_a_template
   end)
 
   def v_datastore_path
-    s = storage
-    datastorepath = location || ""
-    datastorepath = "#{s.name}/#{datastorepath}"  unless s.nil?
-    datastorepath
+    return "#{storage.name}/#{datastorepath}" if storage
+    location || ""
```

Comment:
I'm not sure these are equivalent...

In the before case wouldn't we end up returning `"#{storage.name}/#{location}"` in the case where storage and location are both non-nil?

In the after case it doesn't look like `location` can ever get into the return value even if `storage` is non-nil.

Can you add specs for this before refactoring?